lib/model/content.dart

```diff
@@ -45,6 +52,14 @@ class UnimplementedBlockContentNode extends BlockContentNode
 // A `br` element.
 class LineBreakNode extends BlockContentNode {
   const LineBreakNode({super.debugHtmlNode});
+
+  @override
+  bool operator ==(Object other) {
+    return other is LineBreakNode;
```

Comment:
`LineBreakNode` is unlikely to be subclassed, I think, but isn't that a factor in choosing this implementation with [`is`](https://dart.dev/guides/language/language-tour#type-test-operators) over one that equality-checks the `runtimeType`s? The [`operator ==` example in the Flutter project's style guide](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#common-boilerplates-for-operator--and-hashcode) compares `runtimeTypes`:

```dart
@override
bool operator ==(Object other) {
  if (other.runtimeType != runtimeType) {
    return false;
  }
  // […]
}
```

If we have a class `Animal` that overrides `==`, and its subclasses `Dog` and `Cat` that don't, then `someDog == someCat` will give
- true if `Animal`'s `==` override is `other is Animal`
- false if `Animal`'s `==` override is `other.runtimeType == runtimeType`

(False is probably the right answer.)

<details>
<summary>(code that you can paste into dartpad.dev)</summary>

```dart
class Animal {
  @override
  bool operator ==(Object other) {
    return other is Animal;
  }
  
  @override
  int get hashCode => 'Animal'.hashCode;
}

class Dog extends Animal {}
class Cat extends Animal {}

Dog someDog = Dog();
Cat someCat = Cat();

main() {
  print('someDog == someCat: ${someDog == someCat}');
}

```
</details>

Reply:
Am I being too paranoid? 😅

Reply:
Oh, and:

> `LineBreakNode` is unlikely to be subclassed, I think

Is this right? If so, can I make the same judgment of the other non-`abstract` `FooNode` classes in the file (so e.g. `LineBreakInlineNode` but not `EmojiNode`)?

Reply:
Yeah, none of these classes should ever have subclasses beyond the ones defined in this file. (So in Dart 3, we'll declare them as `sealed`: https://github.com/dart-lang/language/blob/a374667bcc4095edd22e2e6d454cbe9d9f9d85fa/accepted/future-releases/sealed-types/feature-specification.md .) For most of them, including `LineBreakNode`, that means they'll have no subclasses at all.